src/aznfswatchdog

```diff
@@ -384,5 +437,196 @@ while :; do
             fi
         fi
     done
+}
+
+#
+# Watchdog for monitoring unmounts and stunnel processes
+#
+processNFS4Mounts()
+{
+    epoch_now=$(date +%s)
 
+    #
+    # Go over all lines in $AZ_FILES_MOUNTMAP and check them for two things:
+    # 1. Is that entry still in use by at least one aznfs mount, if not remove the entry.
+    # 2. Is stunnel process running?
+    #
+    # We store the mtime of AZ_FILES_MOUNTMAP while inside the lock so that if any mount helper process
+    # updates it after this we will skip modification for sake of safety. We will come to it
+    # in the next iteration when it's safer.
+    #
+    exec {fd}<$AZ_FILES_MOUNTMAP
+    flock -e $fd
+    mtime_mountmap=$(stat -c%Y $AZ_FILES_MOUNTMAP)
+    IFS=$'\n' lines=$(cat $AZ_FILES_MOUNTMAP)
+    flock -u $fd
+    exec {fd}<&-
+
+    #
+    # findmnt must be done after reading AZ_FILES_MOUNTMAP so that if we come accross a
+    # AZ_FILES_MOUNTMAP entry whose all nfs file shares are unmounted, we know
+    # for sure that it's not in use by any mount and can be removed.
+    #
+    findmnt=$(findmnt | grep 'nfs4\|$LOCALHOST' 2>&1)
+
+    #
+    # For no matching mounts also, findmnt exits with a failure return, so check
+    # for both exit status and non-empty error o/p.
+    #
+    if [ $? -ne 0 -a -n "$findmnt" ]; then
+        eecho "${findmnt}."
+        eecho "[FATAL] findmnt failed unexpectedly!"
+        eecho "[FATAL] Azfilenfs-watchdog service is exiting, will not monitor Azure NFS file shares."
+        eecho "[FATAL] Please contact Microsoft support before using any NFS File shares."
+        # This usually indicates some non-transient issue, bail out.
+        exit 1
+    fi
+
+    for line in $lines; do
+        if [ -z "$line" ]; then
+            continue
```

Comment:
where are we remove the empty line?

Reply:
There won't be multiple empty lines. Just one empty line when we unmount all nfsv4 shares (basically when file has no entry after unmounts). Hope that should be okay.